### PR TITLE
Add alarms for expired inflights

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

Add alarms for expired inflights. 

# TESTING:

To test after its merged into staging

### CODE REVIEW

**Comment by jzbahrai** [comment]
I didn't updated terragrunt/production.hcl as I only want to test these alarms on staging first. Is this correct?

File: aws/common/cloudwatch_alarms.tf [hunk 1]
```diff
@@ -1272,3 +1272,122 @@ resource "aws_cloudwatch_metric_alarm" "bulk-bulk-not-being-processed-critical"
...
+        expired           = "True"
+        notification_type = "sms"
+      }
+    }
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "expired-email-created-warning" {
+  alarm_name          = "expired-email-created-warning"
+  alarm_description   = "Email inflights are expiring at > ${var.alarm_warning_expired_sms_created_threshold} for 5 minutes"
```
Comment (sastels) [inline aws/common/cloudwatch_alarms.tf:1337]: `alarm_warning_expired_sms_created_threshold` -> `alarm_warning_expired_email_created_threshold`

**Review by sastels — CHANGES_REQUESTED** [review]
LGTM - minor comment on the alarm string, and update those google docs

File: aws/common/cloudwatch_alarms.tf (hunk 1 above)
Comment (jzbahrai) [reply aws/common/cloudwatch_alarms.tf]: done

File: aws/common/cloudwatch_alarms.tf [hunk 2]
```diff
@@ -1272,3 +1272,122 @@ resource "aws_cloudwatch_metric_alarm" "bulk-bulk-not-being-processed-critical"
...
 
+resource "aws_cloudwatch_metric_alarm" "expired-sms-created-warning" {
+  alarm_name          = "expired-sms-created-warning"
+  alarm_description   = "SMS inflights are expiring at > ${var.alarm_warning_expired_sms_created_threshold} for 5 minutes"
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = "1"
+  threshold           = var.alarm_warning_expired_sms_created_threshold
+  treat_missing_data  = "notBreaching"
+
+  alarm_actions = [aws_sns_topic.notification-canada-ca-alert-critical.arn]
```
Comment (sastels) [inline aws/common/cloudwatch_alarms.tf:1283]: oh, just noticed, this should be 
```
  alarm_actions = [aws_sns_topic.notification-canada-ca-alert-warning.arn]
```

File: aws/common/cloudwatch_alarms.tf [hunk 3]
```diff
@@ -1272,3 +1272,122 @@ resource "aws_cloudwatch_metric_alarm" "bulk-bulk-not-being-processed-critical"
...
+  ok_actions    = [aws_sns_topic.notification-canada-ca-alert-ok.arn]
```
Comment (sastels) [inline aws/common/cloudwatch_alarms.tf:1284]: and we don't need the `ok_action` for the warning

File: aws/common/cloudwatch_alarms.tf [hunk 4]
```diff
@@ -1272,3 +1272,122 @@ resource "aws_cloudwatch_metric_alarm" "bulk-bulk-not-being-processed-critical"
...
+resource "aws_cloudwatch_metric_alarm" "expired-email-created-warning" {
+  alarm_name          = "expired-email-created-warning"
+  alarm_description   = "Email inflights are expiring at > ${var.alarm_warning_expired_email_created_threshold} for 5 minutes"
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = "1"
+  threshold           = var.alarm_warning_expired_email_created_threshold
+  treat_missing_data  = "notBreaching"
+
+  alarm_actions = [aws_sns_topic.notification-canada-ca-alert-critical.arn]
+  ok_actions    = [aws_sns_topic.notification-canada-ca-alert-ok.arn]
```
Comment (sastels) [inline aws/common/cloudwatch_alarms.tf:1344]: Also here with ` alarm_actions = [aws_sns_topic.notification-canada-ca-alert-warning.arn]` and no warning

**Review by sastels — APPROVED** [review]
:shipit: after adjusting those alarm actions!

File: aws/common/cloudwatch_alarms.tf (hunk 2 above)
Comment (jzbahrai) [reply aws/common/cloudwatch_alarms.tf]: done

File: aws/common/cloudwatch_alarms.tf (hunk 4 above)
Comment (jzbahrai) [reply aws/common/cloudwatch_alarms.tf]: done

File: aws/common/cloudwatch_alarms.tf (hunk 2 above)
Comment (sastels) [reply aws/common/cloudwatch_alarms.tf]: otherwise I assume ops genie will page on every celery error. I'm not on call for a couple weeks, though 😂

**Comment by jimleroyer** [comment]
@jzbahrai Thanks for adding these and making us more aware. I was wondering if we saw cases where the expiration got past the 5 minutes? I am asking as I was wondering if the auto move of the messages on the redis list from inflight to inbox would reset the metrics and prevent us from detecting expired inflights? 🤔

**Comment by jzbahrai** [comment]
Screen Shot 2022-09-16 at 3 54 25 PM

**Pull request merged**